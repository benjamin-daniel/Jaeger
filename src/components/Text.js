import React, {ReactNode} from 'react';
import {createText} from '@shopify/restyle';
import {Text as RNText, TextProps as RNTextProps} from 'react-native';

const _Text = (props) => <RNText {...props} />;

export const Text = createText(_Text);

Text.defaultProps = {
  variant: 'bodyText',
  // color: 'bodyText',
  // fontFamily: 'Telex-Regular',
};
