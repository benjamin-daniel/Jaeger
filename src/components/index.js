export * from './Box';
export * from './ErrorText';
export * from './Input';
