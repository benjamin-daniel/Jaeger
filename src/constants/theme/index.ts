import {createTheme} from '@shopify/restyle';

export const palette = {};

const theme = createTheme({
  colors: {},
  spacing: {
    /* eslint-disable id-length */
    xxs: 3,
    xs: 6,
    s: 8,
    m: 10,
    l: 20,
    xl: 32,
    xxl: 43,
    '-s': -8,
    '-m': -16,
    '-l': -24,
    '-xl': -32,
    '-xxl': -43,
    none: 0,
    /* eslint-enable id-length */
  },
  breakpoints: {
    phone: 0,
  },
  textVariants: {},
  inputVariants: {},
  maxContentWidth: 500,
});

export default theme;
