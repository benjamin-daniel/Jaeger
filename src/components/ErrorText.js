import React from 'react';
import {Text} from './Text';

export const ErrorText = ({error, touched, ErrorTextStyles}) => (
  <>
    {touched && error && (
      <Text fontSize={12} color="error" style={[ErrorTextStyles]}>
        {error}
      </Text>
    )}
  </>
);
