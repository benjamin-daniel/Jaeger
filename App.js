import React from 'react';
import {Text} from 'react-native';
import {ThemeProvider} from '@shopify/restyle';

import theme from 'constants/theme';

const App = () => {
  return (
    <>
      <ThemeProvider theme={theme}>
        <Text>Omo</Text>
      </ThemeProvider>
    </>
  );
};

export default App;
