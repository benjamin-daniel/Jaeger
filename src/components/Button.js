import React from 'react';
import {
  TouchableOpacity,
  ActivityIndicator,
  StyleSheet,
  Platform,
  TextStyle,
  ViewStyle,
} from 'react-native';
import {useTheme} from '@shopify/restyle';

import {Text} from './Text';
import {Box} from './Box';
import {palette} from 'constants/theme';

export const Button = ({
  disabled,
  loading,
  variant = 'primary',
  text,
  onPress,
  color: buttonColorName = palette.white,
  textVariant = 'button',
}) => {
  const theme = useTheme();
  const variantProps = theme.buttonVariants[variant];
  const disabledProps = disabled ? variantProps.disabled || {} : {};
  const themedStyles = {...variantProps, ...disabledProps};
  const {
    fontSize,
    fontWeight,
    fontFamily,
    color,
    borderWidth,
    height,
  } = themedStyles;
  const textColor = themedStyles.textColor;
  const buttonColor = buttonColorName && theme.colors[buttonColorName];

  const onPressHandler = loading ? () => {} : onPress;
  const content = (
    <Box
      borderRadius={5}
      alignItems="center"
      justifyContent="center"
      style={{
        backgroundColor: buttonColor || color,
        minHeight: height,
        borderWidth,
        borderColor: buttonColor,
      }}
      paddingHorizontal="m"
      flexDirection="row">
      {loading ? (
        <ActivityIndicator color={textColor} />
      ) : (
        <>
          <Text
            variant={textVariant}
            style={{
              color: textColor || buttonColor,
              fontWeight,
              fontFamily,
              fontSize,
            }}>
            {text}
          </Text>
        </>
      )}
    </Box>
  );
  return (
    <TouchableOpacity
      onPress={onPressHandler}
      style={[styles.stretch]}
      disabled={loading}
      {...{disabled}}>
      {content}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    paddingVertical: 16,
  },
  stretch: {
    alignSelf: 'stretch',
  },
  content: {
    textAlign: 'center',
  },
});
