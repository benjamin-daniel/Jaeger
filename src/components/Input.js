import React, {useState} from 'react';
import {TextInput, StyleSheet, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';
import Feather from 'react-native-vector-icons/Feather';
import {useTheme} from '@shopify/restyle';
import {Box} from './Box';
import {ErrorText} from './ErrorText';
/**
 *
 * @see https://lefkowitz.me/visual-guide-to-react-native-textinput-keyboardtype-options/
 */

const DEFAULT_STYLE = {
  //   fontFamily: 'Telex-Regular',
};

export const Input = ({
  inputStyle,
  containerStyle,
  touched,
  error,
  passwordIcon,
  variant,
  labelVariant,
  Icon,
  innerContainerProps,
  LeftIcon,
  RightIcon,
  ErrorTextStyles,
  nospace,
  ...props
}) => {
  const [state, setState] = useState({
    icon: 'eye-off',
    password: !!Icon,
  });

  const theme = useTheme();
  const variantProps = theme.inputVariants[variant];
  const ICON_SIZE = 20;
  const realInputStyle = StyleSheet.flatten([
    DEFAULT_STYLE,
    variantProps,
    inputStyle,
  ]);
  const changeIcon = () => {
    setState((prevState) => ({
      icon: prevState.icon === 'eye' ? 'eye-off' : 'eye',
      password: !prevState.password,
    }));
  };
  return (
    <Box marginVertical={nospace ? 'none' : 's'}>
      <Box
        borderWidth={1}
        borderColor={touched && error ? 'error' : 'black'}
        flexDirection="row"
        alignItems="center"
        height={40}
        padding="m"
        backgroundColor="mainBackground"
        borderRadius={5}
        {...innerContainerProps}>
        {LeftIcon}
        <Box flex={1}>
          <TextInput
            style={realInputStyle}
            secureTextEntry={state.password}
            {...props}
          />
        </Box>
        {RightIcon}
        {Icon && (
          <TouchableOpacity onPress={() => changeIcon()}>
            <Box
              marginHorizontal="m"
              alignItems="center"
              justifyContent="center">
              <Feather size={ICON_SIZE} name={state.icon} />
            </Box>
          </TouchableOpacity>
        )}
      </Box>
      <ErrorText {...{touched, error, ErrorTextStyles}} />
    </Box>
  );
};
