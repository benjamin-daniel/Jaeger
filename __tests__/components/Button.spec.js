/**
 * @format
 */

import 'react-native-gesture-handler';
import 'react-native';
import React from 'react';
import {Container} from '../../App';
import {palette} from 'constants/theme';
import {fireEvent, render} from 'react-native-testing-library';

import {Button} from 'components';
import renderer, {act} from 'react-test-renderer';
import {
  TouchableOpacity,
  View,
  ActivityIndicator,
} from 'react-native-gesture-handler';

describe('Button component test', () => {
  it('render correctly', async () => {
    await act(async () =>
      renderer.create(
        <Container>
          <Button />
        </Container>,
      ),
    );
  });
  it('match snapshot', async () => {
    const tree = renderer
      .create(
        <Container>
          <Button text="wahala" />
        </Container>,
      )
      .toJSON();
    await expect(tree).toMatchSnapshot();
  });

  it('shows loader when loading', async () => {
    const {root} = renderer.create(
      <Container>
        <Button loading text="wahala" />
      </Container>,
    );
    const loading = root.findByType(View);

    console.log({loading});
  });
});
