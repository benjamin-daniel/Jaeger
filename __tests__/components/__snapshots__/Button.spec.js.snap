// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test match snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 10,
        },
        Object {
          "backgroundColor": "#2B44FF",
          "borderColor": undefined,
          "borderWidth": undefined,
          "minHeight": 50,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "display": "flex",
            "fontSize": 20,
            "fontWeight": "bold",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontWeight": undefined,
          },
        ]
      }
    >
      wahala
    </Text>
  </View>
</View>
`;
